### PR TITLE
Several more HttpListener fixes on Unix

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/20231
Fixes https://github.com/dotnet/corefx/issues/20230
Fixes https://github.com/dotnet/corefx/issues/20229
Fixes https://github.com/dotnet/corefx/issues/20228
Fixes https://github.com/dotnet/corefx/issues/20100
Fixes https://github.com/dotnet/corefx/issues/19977
Fixes https://github.com/dotnet/corefx/issues/20239
Fixes https://github.com/dotnet/corefx/issues/20237
Fixes https://github.com/dotnet/corefx/issues/20101

cc: @geoffkizer, @janvorli, @hughbe, @Priya91, @steveharter

### CODE REVIEW

File: src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerResponse.Managed.cs [hunk 1]
```diff
@@ -63,15 +63,16 @@ public Version ProtocolVersion
...
                     throw new ArgumentNullException(nameof(value));
-
+                }
                 if (value.Major != 1 || (value.Minor != 0 && value.Minor != 1))
+                {
                     throw new ArgumentException(SR.net_wrongversion, nameof(value));
+                }
 
-                _version = value;
+                _version = new Version(value.Major, value.Minor);
```
Comment (geoffkizer) [inline src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerResponse.Managed.cs]: Why?  To drop the other values e.g. Revision?  If so can we add a comment?

Comment (stephentoub) [inline src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerResponse.Managed.cs]: To match the Windows behavior.  I can add a comment.

File: src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs [hunk 2]
```diff
@@ -238,24 +253,31 @@ internal void AddHeader(string header)
 
             string name = header.Substring(0, colon).Trim();
             string val = header.Substring(colon + 1).Trim();
-            string lower = name.ToLower(CultureInfo.InvariantCulture);
-            _headers.Set(name, val);
-            switch (lower)
+            if (name.Equals("content-length", StringComparison.InvariantCultureIgnoreCase))
```
Comment (justinvp) [inline src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs]: StringComparison.OrdinalIgnoreCase?

File: src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs [hunk 3]
```diff
@@ -238,24 +253,31 @@ internal void AddHeader(string header)
...
+                {
+                    _context.ErrorMessage = "Invalid Content-Length.";
+                }
+                else
+                {
+                    _contentLength = parsedContentLength;
+                    _clSet = true;
+                }
+            }
+            else if (name.Equals("transfer-encoding", StringComparison.InvariantCultureIgnoreCase))
```
Comment (justinvp) [inline src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs]: StringComparison.OrdinalIgnoreCase?

**Comment by geoffkizer** [comment]
Overall LGTM

File: src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs]: Sure, I can change that.

File: src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs (hunk 3 above)
Comment (stephentoub) [inline src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs]: Sure

**Pull request closed without merging**